line_detection_updated.py

```diff
@@ -0,0 +1,302 @@
+        if self.mask is True:
+            x_top, x_bottom, y_left, y_right = image_mask(thresholded_image, self.mask_percent)
+
+            thresholded_image = thresholded_image[x_top:x_bottom, y_left:y_right]
+
+        # Deciding Kernel size of blur based on amount of noise
+        num_bright_pixels = np.sum(thresholded_image > np.mean(thresholded_image))
+        x = thresholded_image.shape[0]
+        y = thresholded_image.shape[1]
+        prop_bright_pixels = num_bright_pixels/(x*y)
+
+        for i in range(len(self.flux_prop_thresholds)):
+            if prop_bright_pixels < self.flux_prop_thresholds[i]:
+                kernel_blur = self.blur_kernel_sizes[i]
+                break
+
+        # Blur the image
+        blurred_image = cv2.medianBlur(thresholded_image, kernel_blur)
+
+        # Performing Canny edge detection
+        edges = cv2.Canny(blurred_image, 0, 200)
+
+        return (thresholded_image, blurred_image, edges)
+
+    def hough_transformation(self):
+        '''
+        Detects the straight lines in an image
+
+        Parameters
+        ----------
+        self : 'class'
+            The LineDetection class to use its __init__ parameters
+
+        Returns
+        -------
+        lines: 'numpy.ndarray'
+            The polar coordinates of detected lines
+        angles_list: 'list'
+            List of angles (in radians) for each detected line
+        cart_coords_list: 'list'
+            List of endpoint cartesian coordinates for each detected line
+        thresholded_image: 'numpy.ndarray'
+            The thresholded image from process_image returns
+        blurred_image: 'numpy.ndarray'
+            The blurred image from process_image returns
+        edges: 'numpy.ndarray'
+            The Canny edge detected image from process_image returns
+        '''
+
+        # Processing the image
+        thresholded_image, blurred_image, edges = self.process_image()
+
+        # Performing Hough transformation
+        dimx, dimy = self.image.shape
+        diagonal = np.sqrt(dimx**2 + dimy**2)
+        thresh = int(self.threshold * diagonal)
+        tested_angles = np.linspace(-np.pi / 2, np.pi / 2, 360, endpoint=False)
+
+        # Applying Hough lines to retrieve all possible lines
+        h, theta, d = hough_line(edges, theta=tested_angles)
+
+        # Retriving the peaks
+        accum, angles, dists = hough_line_peaks(h, theta, d, threshold=thresh)
+        print(f"Found {len(dists)} lines.")
+
+        # Finding the cartesian coordinates and storing the returns
+        lines = np.vstack((dists, angles)).T
+        cart_coords_list = []
+        angles_list = []
+
+        for i in range(len(angles)):
+            (x0, y0) = dists[i] * np.array([np.cos(angles[i]), np.sin(angles[i])])
+            cart_coords_list.append((x0, y0))
+            angles_list.append(angles[i])
+
+        return lines, angles_list, cart_coords_list, thresholded_image, \
+            blurred_image, edges
```

Comment:
This is a lot of return values, it might benefit from actually returning a dictionary of these values, in that way it's self-documenting on what is what, or like an "result" object or something with these attributes. I'm ok with this for now though, we can address it during integration.